Two tests/jose/test_jwe.py tests failing

While packaging this package for openSUSE we try to start running the testsuite during the packaging (so that we may catch some unexpected failure to build package correctly) and when running `tests/jose` I got this:

```
[   11s] + python3.9 -mpytest tests/jose
[   11s] ============================= test session starts ==============================
[   11s] platform linux -- Python 3.9.12, pytest-7.1.1, pluggy-1.0.0
[   11s] rootdir: /home/abuild/rpmbuild/BUILD/authlib-1.0.1, configfile: tox.ini
[   11s] plugins: anyio-3.5.0, asyncio-0.17.2
[   11s] asyncio: mode=auto
[   11s] collected 134 items
[   11s]
[   11s] tests/jose/test_jwe.py ..................F.............................. [ 36%]
[   11s] ............................F                                            [ 58%]
[   12s] tests/jose/test_jwk.py .......................                           [ 75%]
[   12s] tests/jose/test_jws.py ................                                  [ 87%]
[   12s] tests/jose/test_jwt.py .................                                 [100%]
[   12s]
[   12s] =================================== FAILURES ===================================
[   12s] __________________________ JWETest.test_dir_alg_xc20p __________________________
[   12s]
[   12s] self = <tests.jose.test_jwe.JWETest testMethod=test_dir_alg_xc20p>
[   12s]
[   12s]     def test_dir_alg_xc20p(self):
[   12s]         jwe = JsonWebEncryption()
[   12s]         key = OctKey.generate_key(256, is_private=True)
[   12s]         protected = {'alg': 'dir', 'enc': 'XC20P'}
[   12s] >       data = jwe.serialize_compact(protected, b'hello', key)
[   12s]
[   12s] tests/jose/test_jwe.py:2657:
[   12s] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
[   12s] authlib/jose/rfc7516/jwe.py:80: in serialize_compact
[   12s]     enc = self.get_header_enc(protected)
[   12s] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
[   12s]
[   12s] self = <authlib.jose.rfc7516.jwe.JsonWebEncryption object at 0x7fad22f4ac40>
[   12s] header = {'alg': 'dir', 'enc': 'XC20P'}
[   12s]
[   12s]     def get_header_enc(self, header):
[   12s]         if 'enc' not in header:
[   12s]             raise MissingEncryptionAlgorithmError()
[   12s]         enc = header['enc']
[   12s]         if self._algorithms and enc not in self._algorithms:
[   12s]             raise UnsupportedEncryptionAlgorithmError()
[   12s]         if enc not in self.ENC_REGISTRY:
[   12s] >           raise UnsupportedEncryptionAlgorithmError()
[   12s] E           authlib.jose.errors.UnsupportedEncryptionAlgorithmError: unsupported_encryption_algorithm: Unsupported "enc" value in header
[   12s]
[   12s] authlib/jose/rfc7516/jwe.py:678: UnsupportedEncryptionAlgorithmError
[   12s] _______________ JWETest.test_xc20p_content_encryption_decryption _______________
[   12s]
[   12s] self = <tests.jose.test_jwe.JWETest testMethod=test_xc20p_content_encryption_decryption>
[   12s]
[   12s]     def test_xc20p_content_encryption_decryption(self):
[   12s]         # https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-xchacha-03#appendix-A.3.1
[   12s] >       enc = JsonWebEncryption.ENC_REGISTRY['XC20P']
[   12s] E       KeyError: 'XC20P'
[   12s]
[   12s] tests/jose/test_jwe.py:2672: KeyError
[   12s] =========================== short test summary info ============================
[   12s] FAILED tests/jose/test_jwe.py::JWETest::test_dir_alg_xc20p - authlib.jose.err...
[   12s] FAILED tests/jose/test_jwe.py::JWETest::test_xc20p_content_encryption_decryption
[   12s] ======================== 2 failed, 132 passed in 1.08s =========================
```

[Complete build log](https://github.com/lepture/authlib/files/8655846/_log.txt) with all details of packages used and steps taken to reproduce.


 - OS: openSUSE/Tumbleweed as of 2022-05-10
 - Python Version: various versions of Python, this traceback is from 3.9.12, pytest-7.1.1, and pluggy-1.0.0
 - Authlib Version: 1.0.1

I am **packaging** 1.5.2, so yes, I do (you can see versions of all packages used and steps taken in the attached log).

```
[   37s] =================================== FAILURES ===================================
[   37s] _______________________ ChaCha20Test.test_dir_alg_xc20p ________________________
[   37s] 
[   37s] self = <tests.jose.test_chacha20.ChaCha20Test testMethod=test_dir_alg_xc20p>
[   37s] 
[   37s]     def test_dir_alg_xc20p(self):
[   37s]         jwe = JsonWebEncryption()
[   37s]         key = OctKey.generate_key(256, is_private=True)
[   37s]         protected = {"alg": "dir", "enc": "XC20P"}
[   37s] >       data = jwe.serialize_compact(protected, b"hello", key)
[   37s] 
[   37s] tests/jose/test_chacha20.py:28: 
[   37s] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[   37s] authlib/jose/rfc7516/jwe.py:91: in serialize_compact
[   37s]     enc = self.get_header_enc(protected)
[   37s] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[   37s] 
[   37s] self = <authlib.jose.rfc7516.jwe.JsonWebEncryption object at 0x7fc3583cebd0>
[   37s] header = {'alg': 'dir', 'enc': 'XC20P'}
[   37s] 
[   37s]     def get_header_enc(self, header):
[   37s]         if "enc" not in header:
[   37s]             raise MissingEncryptionAlgorithmError()
[   37s]         enc = header["enc"]
[   37s]         if self._algorithms is not None and enc not in self._algorithms:
[   37s]             raise UnsupportedEncryptionAlgorithmError()
[   37s]         if enc not in self.ENC_REGISTRY:
[   37s] >           raise UnsupportedEncryptionAlgorithmError()
[   37s] E           authlib.jose.errors.UnsupportedEncryptionAlgorithmError: unsupported_encryption_algorithm: Unsupported 'enc' value in header
[   37s] 
[   37s] authlib/jose/rfc7516/jwe.py:713: UnsupportedEncryptionAlgorithmError
[   37s] ____________ ChaCha20Test.test_xc20p_content_encryption_decryption _____________
[   37s] 
[   37s] self = <tests.jose.test_chacha20.ChaCha20Test testMethod=test_xc20p_content_encryption_decryption>
[   37s] 
[   37s]     def test_xc20p_content_encryption_decryption(self):
[   37s]         # https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-xchacha-03#appendix-A.3.1
[   37s] >       enc = JsonWebEncryption.ENC_REGISTRY["XC20P"]
[   37s] E       KeyError: 'XC20P'
[   37s] 
[   37s] tests/jose/test_chacha20.py:39: KeyError
[   37s] =========================== short test summary info ============================
[   37s] FAILED tests/jose/test_chacha20.py::ChaCha20Test::test_dir_alg_xc20p - authli...
[   37s] FAILED tests/jose/test_chacha20.py::ChaCha20Test::test_xc20p_content_encryption_decryption
[   37s] ======================== 2 failed, 139 passed in 0.90s =========================
```

[Complete build log](https://github.com/user-attachments/files/19996706/_log.txt)

--- a/tests/jose/test_chacha20.py
+++ b/tests/jose/test_chacha20.py
@@ -1,5 +1,7 @@
 import unittest
 
+import pytest
+
 from authlib.jose import JsonWebEncryption
 from authlib.jose import OctKey
 from authlib.jose.drafts import register_jwe_draft
@@ -22,6 +24,8 @@ class ChaCha20Test(unittest.TestCase):
         self.assertRaises(ValueError, jwe.serialize_compact, protected, b"hello", key2)
 
     def test_dir_alg_xc20p(self):
+        pytest.importorskip("Cryptodome.Cipher.ChaCha20_Poly1305")
+
         jwe = JsonWebEncryption()
         key = OctKey.generate_key(256, is_private=True)
         protected = {"alg": "dir", "enc": "XC20P"}
@@ -35,6 +39,8 @@ class ChaCha20Test(unittest.TestCase):
         self.assertRaises(ValueError, jwe.serialize_compact, protected, b"hello", key2)
 
     def test_xc20p_content_encryption_decryption(self):
+        pytest.importorskip("Cryptodome.Cipher.ChaCha20_Poly1305")
+
         # https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-xchacha-03#appendix-A.3.1
         enc = JsonWebEncryption.ENC_REGISTRY["XC20P"]
 

